### ISSUE TITLE
Yamlfixer hangs on Git bash

### ISSUE THREAD

**Issue by adriens** [opt-nc/yamlfixer#43]
![image](https://user-images.githubusercontent.com/5235127/158725416-ef27e1bf-31b1-4ffb-820a-78bd194ea6dd.png)

**Comment by tamere-allo-peter** [comment]
Looks like a bug in git bash for me. If yamlfixer hangs its probably because it tries to read input from stdin, and it only does this if : 
* either `-` is specified as a filename on the command line, which is not the case here
* or `stdin` is not a TTY, i.e. stdin is a pipe, which should not be the case here, unless git bash is broken.

**Comment by tamere-allo-peter** [comment]
@adriens : if pressing `Ctrl+D` cleanly interrupts yamlfixer's hanging, see case 2 above.

**Comment by adriens** [comment]
>  if pressing Ctrl+D cleanly interrupts yamlfixer's hanging, see case 2 above.

It does not interrupt, even not `^C`

**Comment by tamere-allo-peter** [comment]
Strange... Could you launch again with --debug then ?

**Comment by adriens** [comment]
![image](https://user-images.githubusercontent.com/5235127/158731539-256c2e62-562f-4ab6-9d5c-777449c2fd0c.png)

**Comment by tamere-allo-peter** [comment]
That's what I said : you shell is buggy. In the code there's this : 

```python
# Ensure we read from stdin in case it's redirected                                                                                                        
if ("-" not in sys.argv[1:]) and not sys.stdin.isatty():
    sys.argv.append("-")
```
This condition should not be met, but it is, hence there's probably a bug in your shell.

**Comment by tamere-allo-peter** [comment]
@adriens could you paste the result of this on your git bash shell : 

```
jerome@OPT17844:~$ python3
Python 3.8.10 (default, Nov 26 2021, 20:14:08) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.stdin.isatty()
True
>>> 
```

**Comment by adriens** [comment]
I'll provide yo uthat this afternoon 👍

**Comment by adriens** [comment]
![image](https://user-images.githubusercontent.com/5235127/158735485-a4ff02db-7881-46a0-9f3c-30fd2909fb7f.png)

**Comment by tamere-allo-peter** [comment]
This sucks...

**Comment by adriens** [comment]
Sure 💩 💋

**Comment by tamere-allo-peter** [comment]
Big time !

**Comment by adriens** [comment]
Legendary 💣

**Comment by tamere-allo-peter** [comment]
@adriens Not a bug in `yamlfixer` for sure, please see https://github.com/git-for-windows/git/issues/2594

I'll keep this one open though, because there's a similar issue when run from a GitHub action and I'm working on this atm.

**Comment by tamere-allo-peter** [comment]
@adriens here's a workaround for you : https://jpcodeqa.com/q/37d31f92ab77ec9de6be7e217da60985

**Comment by adriens** [comment]
>  there's a similar issue when run from a GitHub action and I'm working on this atm.

in a way that's a good news you could reproduce it 🤞

**Comment by adriens** [comment]
@mbarre  pour `Dockerfile`, les variables à positionner

```python
  and not os.environ.get("GITHUB_ACTIONS") \
  and not os.environ.get("CI"):
```

exposer les variables dans `Dockerfile` et les positionner à `true`

**Comment by mbarre** [comment]
Yep, je viens d'ajouter GITHUB_ACTIONS, publication de l'image docker en cours

**Comment by adriens** [comment]
cf https://github.com/opt-nc/yamlfixer/blob/99420745922981c9a7403e22de88cddbcfd58f6f/yamlfixer/__main__.py#L48
